.github/workflows/build_commits.yaml

```diff
@@ -0,0 +1,58 @@
+name: build_commits
+
+on: [pull_request]
+
+jobs:
+  build_commits:
+    name: Check if build works for every commit
+    runs-on: ubuntu-latest
+    timeout-minutes: 60
+    steps:
+      - name: Configure git
+        run: |
+          git config --global user.name "GitHub Actions"
+          git config --global user.email "github-actions@users.noreply.github.com"
+
+      - name: Install Go
+        uses: actions/setup-go@37335c7bb261b353407cff977110895fa0b4f7d8
+        with:
+          go-version: 1.16.3
+          
+      - name: Cache LLVM and Clang
+        id: cache-llvm
+        uses: actions/cache@26968a09c0ea4f3e233fdddbafd1166051a095f6
+        with:
+          path: $HOME/.clang
+          key: llvm-10.0
+          
+      - name: Install LLVM and Clang
+        uses: KyleMayes/install-llvm-action@4609a7b44dbb15353bdb46dafaac2914948399c9
+        with:
+          version: "10.0"
+          directory: $HOME/.clang
+          cached: ${{ steps.cache-llvm.outputs.cache-hit }}
+          
+      - name: Install libelf
+        run: |
+          sudo apt-get install libelf-dev
+          
+      - name: Checkout code
+        uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f
+        with:
+          ref: ${{ github.event.pull_request.head.sha }}
+          fetch-depth: 0
+          
+      - name: Check if build works for every commit
+        run: |
+          PR_COMMITS_API_JSON=$(curl \
+            -H "Accept: application/vnd.github.v3+json" \
+            -H "Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}" \
+            ${{ github.event.pull_request.commits_url }})
+          PR_FIRST_SHA=$(echo "$PR_COMMITS_API_JSON" | jq -r ".[0].sha")
+          PR_PARENT_SHA=$(git rev-parse "${PR_FIRST_SHA}^")
+          git rebase --exec "make clean && make -j $(nproc) && make -C bpf build_all -j $(nproc)" $PR_PARENT_SHA
```

Comment:
Do we need to run `make clean` here? Also wouldn't `make build` be enough to test if the commits work independently?

Reply:
Yes, I think we should not need `make clean` in Github actions.

Reply:
Aren't we basically running the content of `exec` flag for every commit in PR? In that case shouldn't we leave `make clean`?

Reply:
@nebril Why? If we build commit n, then commit n+1 make changes to a set of files and we rebuild, those files and their dependencies will be re-compiled. Any error introduced by commit n+1 will be caught even though we're not building from scratch.

Reply:
@pchaigno  So the final change will be: `git rebase --exec "make build -j $(nproc)" $PR_PARENT_SHA ` ?

Reply:
> Why drop the `make -C bpf build_all` part?

I thought `build` included all the sub-directories as the target, including 'bpf'.

> André was proposing `make build` in place of `make`, but we can keep the additional `make -C build_all` afterward I think. `build` as a target is just better than the implicit `all` target here because it doesn't build e.g. the docs which we don't need.

Got it now! :+1:

Reply:
> I thought `build` included all the sub-directories as the target, including 'bpf'.

`-C bpf build_all` is a bit unconventional because it's actually a set of compile-tests. That is, it compiles our BPF datapath with a bunch of different combinations of options.